content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1052 @@
+
+![clusters](clusters.png)
+
+The current implementation is limited to at most 256 lights as we initially prioritized cross-platform compatibility so that everyone could benefit. WebGL2 specifically does not support storage buffers and so the implementation is currently constrained by the maximum uniform buffer size. We can support many more lights on other platforms by using storage buffers, which we will add support for in a future release.
+
+[Click here](https://youtu.be/dElYzzNovEk) for a video that illustrates Bevy's clustered forward rendering.
+
+### Sprite Batching
+
+<div class="release-feature-authors">authors: @cart</div>
+
+Sprites are now rendered in batches according to their texture within a z-level. They are also opportunistically batched across z-levels. This yields significant performance wins because it drastically reduces the number of draw calls required. Combine that with the other performance improvements in the new Bevy Renderer and things start to get very interesting! On my machine, the old Bevy renderer generally started dropping below 60fps at around 8,000 sprites in our "bevymark" benchmark. With the new renderer on that same machine I can get about 100,000 sprites!
+
+![bevymark](bevymark.png)
+
+My machine: Nvidia GTX 1070, Intel i7 7700k, 16GB ram, Arch Linux
+
+### Sprite Ergonomics
+
+Sprite entities are now simpler to spawn:
+
+```rust
+fn spawn_sprite(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.spawn_bundle(SpriteBundle {
+        texture: asset_server.load("player.png"),
+        ..Default::default()
+    });
+}
+```
+
+No need to manage sprite materials! Their texture handle is now a direct component and color can now be set directly on the {{rust_type(type="struct" crate="bevy_sprite" version="0.6.0" name="Sprite")}} component.
+
+<details>
+    <summary>To compare, expand this to see the old Bevy 0.5 code</summary>
+
+```rust
+// Old (Bevy 0.5)
+fn spawn_sprite(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    let texture_handle = asset_server.load("player.png");
+    commands.spawn_bundle(SpriteBundle {
+        material: materials.add(texture_handle.into()),
+        ..Default::default()
+    });
+}
+```
+
+</details>
+
+### WGSL Shaders
+
+Bevy now uses [WGSL](https://www.w3.org/TR/WGSL/) for our built-in shaders and examples. WGSL is a new shader language being developed for WebGPU (although it is a "cross platform" shader language just like GLSL). Bevy still supports GLSL shaders, but WGSL is nice enough that, for now, we are treating it as our "officially recommended" shader language. WGSL is still being developed and polished, but given how much investment it is receiving I believe it is worth betting on. Consider this the start of the "official Bevy shader language" conversation, not the end of it.
```

Comment:
```suggestion
Bevy now uses [WGSL](https://www.w3.org/TR/WGSL/) for our built-in shaders and examples. WGSL is a new shader language for WebGPU (although it is a "cross platform" shader language just like GLSL). Bevy still supports GLSL shaders, but WGSL is nice enough that, for now, we are treating it as our "officially recommended" shader language. WGSL is still being developed and polished, but given how much investment it is receiving I believe it is worth betting on. Consider this the start of the "official Bevy shader language" conversation, not the end of it.
```

Reply:
I think `being developed` helps capture the "this is still a work in progress" and "this isn't _just_ for WebGPU" ideas, so I think I'll leave this as-is.

Reply:
That's a fair point. I just suggested this because it makes it easier to read and the sentence after the next is all about WGSL still being developed.